From a73355c9edfbf13734092f7ab89ebe0b507eda09 Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@marvell.com>
Date: Thu, 3 Oct 2019 13:08:33 -0700
Subject: [PATCH 437/767] scmi: driver: insufficient buffer size for rx

commit 4e8f9f6401e5f7c0108333d8dd4b6b73972a341d from
git@git.assembla.com:cavium/WindRiver.linux.git

scmi cpu-freq driver allows MAX_OPPS (16) OPPs but when transferring
the 16 OPP data from SCP to ARM, because of the max_msg_size limited
to 128, it fails to convey the information about all 16 OPPs.

An increase to 256 for max_msg_size accommodates all entries correctly
and linux receives them with integrity.

Bug: IPBUSW-2243

Change-Id: I089d0be50a06ffaa3bc89691297a6e6159bb8e2c
Signed-off-by: Sujeet Baranwal <sbaranwal@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18607
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/firmware/arm_scmi/driver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/firmware/arm_scmi/driver.c b/drivers/firmware/arm_scmi/driver.c
index c910ab489518..0f8d58af8d4a 100644
--- a/drivers/firmware/arm_scmi/driver.c
+++ b/drivers/firmware/arm_scmi/driver.c
@@ -608,7 +608,7 @@ int scmi_handle_put(const struct scmi_handle *handle)
 static const struct scmi_desc scmi_generic_desc = {
 	.max_rx_timeout_ms = 300,	/* We may increase this if required */
 	.max_msg = 20,		/* Limited by MBOX_TX_QUEUE_LEN */
-	.max_msg_size = 128,
+	.max_msg_size = 256,
 };
 
 /* Each compatible listed below must have descriptor associated with it */
-- 
2.31.1

